### PR TITLE
adding defaults for chef_omnibus_root for windows and unix-like/linux systems

### PR DESCRIPTION
**Pull request by tarcinil**
This work patches #923 so that the issue will stop bugging people, and we can discuss implementing mixlib-install but that has downstream affects against cinc.

### CODE REVIEW

File: lib/kitchen/provisioner/chef_base.rb [hunk 1]
```diff
@@ -44,6 +44,13 @@ module Provisioner
     class ChefBase < Base
       default_config :require_chef_omnibus, true
       default_config :chef_omnibus_url, "https://omnitruck.chef.io/install.sh"
+      default_config :chef_omnibus_root do |provisioner|
+        if provisioner.powershell_shell?
+          "/opt/chef" # I know that we don't need this, but it leaves things flexible.
```
Comment (robbkidd) [inline lib/kitchen/provisioner/chef_base.rb]: Any reason not to set this to the Windows default of `"$env:systemdrive\\opscode\\chef"`?

Comment (tarcinil) [reply lib/kitchen/provisioner/chef_base.rb]: I based that on content that I had seen inside of the spec tests.

spec/kitchen/provisioner/chef_base_spec.rb#L604 which looks to expect the same path as unix-like/linux systems.  I did contemplate using `provisioner.windows_os?` to `"$env:systemdrive\\opscode\\chef"` entry.

**Comment by tarcinil** [comment]
As I am setting the default, anything done in the later functions "should" overload the default_config.  I have been thinking adding additional unit tests that would overload this using lets say Cinc as an example.

File: lib/kitchen/provisioner/chef_base.rb (hunk 1 above)
Comment (lamont-granquist) [reply lib/kitchen/provisioner/chef_base.rb:49]: that should be c:\opscode\chef though not /opt/chef for windows

**Pull request closed without merging**